client/src/components/ObjectStore/Instances/icons.js

```diff
@@ -0,0 +1,9 @@
+/* I can't get this to type properly in type script, so
+   handling it here in JavaScript. I get a variant of this
+   error (https://github.com/FortAwesome/Font-Awesome/issues/12575).
+*/
```

Comment:
Regarding this, I guess you already tried to add a `ts-ignore` comment like [here](https://github.com/galaxyproject/galaxy/blob/45c50c204bca4d60774fb23d57b68d1a12b490d2/client/src/components/TagsMultiselect/Tag.vue#L23) and it didn't work...

However there is another workaround, Laila discovered that, in a Vue setup TS component, if you put the `ts-ignore` comment directly after the imports (as we usually do) the "ignore magic" does not work. You need to move the library.add call somewhere **`after`** the `defineProps`.
```js
imports ...

defineProps<Props>();

//@ts-ignore bad types
library.add(...);

```

This seems to be due to some weird side effects when compiling the TS Vue component.